### ISSUE TITLE
Session was terminated due to FORWARDING_TO_NODE_FAILED

### ISSUE THREAD

**Issue by WarleyGabriel** [zalando/zalenium#593]
I'm not sure if this is related to zalenium but seems to happen only with it.

```
org.openqa.selenium.WebDriverException:` Session [c6e9bac9-9772-40cd-a3ec-dcd52bc4d455] was terminated due to FORWARDING_TO_NODE_FAILED
Build info: version: '3.12.0', revision: '7c6e0b3', time: '2018-05-08T14:04:26.12Z'
System info: host: 'softl020-086', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.13.0-43-generic', java.version: '1.8.0_171'
Driver info: org.openqa.selenium.remote.RemoteWebDriver
Capabilities {acceptInsecureCerts: true, browserName: firefox, browserVersion: 60.0.1, javascriptEnabled: true, moz:accessibilityChecks: false, moz:headless: false, moz:processID: 5259, moz:profile: /tmp/rust_mozprofile.PfSuw2..., moz:useNonSpecCompliantPointerOrigin: false, moz:webdriverClick: true, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, platformVersion: 4.4.0-92-generic, rotatable: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, webdriver.remote.sessionid: c6e9bac9-9772-40cd-a3ec-dcd...}
Session ID: c6e9bac9-9772-40cd-a3ec-dcd52bc4d455

	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:187)
	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:122)
	at org.openqa.selenium.remote.http.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:49)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:543)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:600)
	at org.openqa.selenium.remote.RemoteWebDriver.getCurrentUrl(RemoteWebDriver.java:281)
	at geb.Browser.retrieveCurrentUri(Browser.groovy:511)
	at geb.Browser.go(Browser.groovy:498)
	at geb.Page.to(Page.groovy:266)
	at geb.Browser.via(Browser.groovy:618)
	at geb.Browser.to(Browser.groovy:572)
	at geb.Browser.to(Browser.groovy:550)
	at geb.Browser.to(Browser.groovy:539)
	at geb.spock.GebSpec.methodMissing(GebSpec.groovy:56)
```

We get this error randomly.

Any ideas?

Zalenium Image Version(s): 3.12.0
Docker Version:  18.05.0-ce, build f150324
OS: Ubuntu 16.04.4 LTS
Docker Command to start Zalenium:  
``
 docker run -d -ti --name zalenium -p 4444:4444 \
 -v /var/run/docker.sock:/var/run/docker.sock \
 -v /tmp/videos:/home/seluser/videos \
 -v /home/compartilhada:/tmp/node/home/seluser/.softplan-websigner \
--privileged dosel/zalenium start --timeZone "America/Sao_Paulo" \
--maxTestSessions 100 --desiredContainers 5 \
--seleniumImageName selenium-grid:latest
``

**Comment by diemol** [comment]
Hi @WarleyGabriel,

It is hard to say, can we get more information? Providing the Zalenium logs would be useful.

**Comment by WarleyGabriel** [comment]
Hi @diemol 

Is this log helpful?

```
16:38:38.272 [Matcher thread] INFO  org.openqa.grid.internal.TestSlot - Trying to create a new session on test slot {server:CONFIG_UUID=b5f48040-893c-4f7a-b4ce-a05e00a28da6, seleniumProtocol=WebDriver, zal:tz=America/Sao_Paulo, version=60.0.1, platform=LINUX, acceptSslCerts=true, zal:screenResolution=1920x1080, zal:screen-resolution=1920x1080, browserName=firefox, zal:resolution=1920x1080, maxInstances=1, platformName=LINUX}
16:38:38.276 [Matcher thread] INFO  d.z.e.z.registry.ZaleniumRegistry - Test session with internal key 17da1c53-f451-4e5d-a461-8c48c964abc2 assigned to remote (44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9) after 13 seconds (13943 ms).
16:38:40.232 [Forwarding 17da1c53-f451-4e5d-a461-8c48c964abc2 (int. key, remote not contacted yet.) to http://172.17.0.3:40000/wd/hub at 4:38:38 PM] INFO  d.z.e.z.p.DockerSeleniumRemoteProxy - Test session started with internal key 17da1c53-f451-4e5d-a461-8c48c964abc2 and external key 6347f492-c6e8-4f15-8b0a-5d2faaf6b2a0 assigned to remote 44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9.
16:38:40.236 [Forwarding 17da1c53-f451-4e5d-a461-8c48c964abc2 (int. key, remote not contacted yet.) to http://172.17.0.3:40000/wd/hub at 4:38:38 PM] INFO  d.z.e.z.c.DockerContainerClient - http://172.17.0.3:40000 [bash, -c, start-video]
16:45:51.372 [qtp1622458036-19] ERROR o.o.g.w.s.handler.RequestHandler - cannot forward the request unexpected end of stream on Connection{172.17.0.3:40000, proxy=DIRECT hostAddress=/172.17.0.3:40000 cipherSuite=none protocol=http/1.1}
java.io.IOException: unexpected end of stream on Connection{172.17.0.3:40000, proxy=DIRECT hostAddress=/172.17.0.3:40000 cipherSuite=none protocol=http/1.1}
	at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:205)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
	at okhttp3.RealCall.execute(RealCall.java:77)
	at org.openqa.selenium.remote.internal.OkHttpClient.execute(OkHttpClient.java:105)
	at org.openqa.grid.internal.TestSession.sendRequestToNode(TestSession.java:430)
	at org.openqa.grid.internal.TestSession.forward(TestSession.java:237)
	at org.openqa.grid.web.servlet.handler.RequestHandler.forwardRequest(RequestHandler.java:99)
	at org.openqa.grid.web.servlet.handler.RequestHandler.process(RequestHandler.java:133)
	at org.openqa.grid.web.servlet.DriverServlet.process(DriverServlet.java:86)
	at org.openqa.grid.web.servlet.DriverServlet.doPost(DriverServlet.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.seleniumhq.jetty9.servlet.ServletHolder.handle(ServletHolder.java:860)
	at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
	at io.prometheus.client.filter.MetricsFilter.doFilter(MetricsFilter.java:170)
	at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
	at org.seleniumhq.jetty9.servlet.ServletHandler.doHandle(ServletHandler.java:533)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.seleniumhq.jetty9.security.SecurityHandler.handle(SecurityHandler.java:548)
	at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
	at org.seleniumhq.jetty9.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
	at org.seleniumhq.jetty9.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
	at org.seleniumhq.jetty9.servlet.ServletHandler.doScope(ServletHandler.java:473)
	at org.seleniumhq.jetty9.server.session.SessionHandler.doScope(SessionHandler.java:1564)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
	at org.seleniumhq.jetty9.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
	at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.seleniumhq.jetty9.server.handler.StatisticsHandler.handle(StatisticsHandler.java:169)
	at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
	at org.seleniumhq.jetty9.server.Server.handle(Server.java:530)
	at org.seleniumhq.jetty9.server.HttpChannel.handle(HttpChannel.java:347)
	at org.seleniumhq.jetty9.server.HttpConnection.onFillable(HttpConnection.java:256)
	at org.seleniumhq.jetty9.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
	at org.seleniumhq.jetty9.io.FillInterest.fillable(FillInterest.java:102)
	at org.seleniumhq.jetty9.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
	at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
	at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
	at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
	at org.seleniumhq.jetty9.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
	at org.seleniumhq.jetty9.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
	at org.seleniumhq.jetty9.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException: \n not found: limit=0 content=…
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:227)
	at okhttp3.internal.http1.Http1Codec.readHeaderLine(Http1Codec.java:212)
	at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
	... 57 common frames omitted
16:45:51.386 [Terminate Test Session int id: [17da1c53-f451-4e5d-a461-8c48c964abc2] ext id: [6347f492-c6e8-4f15-8b0a-5d2faaf6b2a0] container: [44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9]] INFO  d.z.e.z.p.DockerSeleniumRemoteProxy - 44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9 AFTER_SESSION command received. Cleaning up node for reuse, used 5 of max 100
16:45:51.395 [Terminate Test Session int id: [17da1c53-f451-4e5d-a461-8c48c964abc2] ext id: [6347f492-c6e8-4f15-8b0a-5d2faaf6b2a0] container: [44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9]] INFO  d.z.e.z.c.DockerContainerClient - http://172.17.0.3:40000 [bash, -c, notify 'Zalenium', 'TEST COMPLETED', --icon=/home/seluser/images/completed.png]
16:45:51.527 [Terminate Test Session int id: [17da1c53-f451-4e5d-a461-8c48c964abc2] ext id: [6347f492-c6e8-4f15-8b0a-5d2faaf6b2a0] container: [44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9]] INFO  d.z.e.z.c.DockerContainerClient - http://172.17.0.3:40000 [bash, -c, stop-video]
16:46:24.538 [Terminate Test Session int id: [17da1c53-f451-4e5d-a461-8c48c964abc2] ext id: [6347f492-c6e8-4f15-8b0a-5d2faaf6b2a0] container: [44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9]] INFO  d.z.e.z.p.DockerSeleniumRemoteProxy - 44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9 Video file copied to: /home/seluser/videos/zalenium_17da1c53_f451_4e5d_a461_8c48c964abc2_firefox_LINUX_20180525164621916_COMPLETED.mp4
16:46:24.548 [Terminate Test Session int id: [17da1c53-f451-4e5d-a461-8c48c964abc2] ext id: [6347f492-c6e8-4f15-8b0a-5d2faaf6b2a0] container: [44e72667620ed9d6c9c2681ffc14ea3566cb49be0875f39c9ce490d2e9e180f9]] INFO  d.z.e.z.c.DockerContainerClient - http://172.17.0.3:40000 [bash, -c, transfer-logs.sh
```
Thanks!

**Comment by pearj** [comment]
Hi @WarleyGabriel,

I see in your start command you’re using: selenium-grid:latest

Only images based off `elgalu/selenium` are supported.

**Comment by WarleyGabriel** [comment]
Hi all,

@diemol Yes, this image is based on `elgalu/selenium`, this is DockerFile her.

```
FROM elgalu/selenium

USER root

RUN apt-get update && apt-get install -y pkg-config sudo \
    && wget -q https://artifactory.softplan.com.br/artifactory/common-files-unj/extensions/websigner/softplan-websigner_2.5.2-1_64.deb -O /tmp/websigner.deb \
    && dpkg -i /tmp/websigner.deb \
    && rm /tmp/websigner.deb

USER seluser
```

I get that error after more or less seven minutes of suite execution, usually.
During the test suite the session is created only once, on the first test.
I think the hardware configurations meet, because with the version: 3.11.0e is working.
I can run in `elgalu/selenium` image, if necessary.
Thanks!

**Comment by diemol** [comment]
Just one more question, is this happening during test execution or only when the session is being created?

**Comment by WarleyGabriel** [comment]
Hi, no problem. 
This is happening during test execution, after more or less seven minutes the start.
Thanks!

**Comment by WarleyGabriel** [comment]
Ok, done.
I report any news. 
Thanks!

**Comment by yuandanxu** [comment]
I will have a try too, thanks a lot Diemol.

**Comment by yuandanxu** [comment]
@diemol I just pulled latest selenium and have a test. I encountered following issue after add this parameter:
21:28:41.359 [qtp1829217853-26] INFO  o.o.g.w.s.handler.RequestHandler - Error forwarding the new session Request timed out waiting for a node to become available.
java.util.concurrent.TimeoutException: Request timed out waiting for a node to become available.
        at org.openqa.grid.web.servlet.handler.RequestHandler.waitForSessionBound(RequestHandler.java:197)
        at org.openqa.grid.web.servlet.handler.RequestHandler.process(RequestHandler.java:112)
        at org.openqa.grid.web.servlet.DriverServlet.process(DriverServlet.java:86)
        at org.openqa.grid.web.servlet.DriverServlet.doPost(DriverServlet.java:70)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.seleniumhq.jetty9.servlet.ServletHolder.handle(ServletHolder.java:860)
        at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
        at io.prometheus.client.filter.MetricsFilter.doFilter(MetricsFilter.java:170)
        at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at org.seleniumhq.jetty9.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.seleniumhq.jetty9.security.SecurityHandler.handle(SecurityHandler.java:548)
        at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
        at org.seleniumhq.jetty9.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
        at org.seleniumhq.jetty9.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
        at org.seleniumhq.jetty9.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.seleniumhq.jetty9.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
        at org.seleniumhq.jetty9.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.seleniumhq.jetty9.server.handler.StatisticsHandler.handle(StatisticsHandler.java:169)
        at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.seleniumhq.jetty9.server.Server.handle(Server.java:530)
        at org.seleniumhq.jetty9.server.HttpChannel.handle(HttpChannel.java:347)
        at org.seleniumhq.jetty9.server.HttpConnection.onFillable(HttpConnection.java:256)
        at org.seleniumhq.jetty9.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
        at org.seleniumhq.jetty9.io.FillInterest.fillable(FillInterest.java:102)
        at org.seleniumhq.jetty9.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
        at org.seleniumhq.jetty9.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
        at org.seleniumhq.jetty9.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:708)
        at org.seleniumhq.jetty9.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:626)
        at java.lang.Thread.run(Thread.java:748)

**Comment by diemol** [comment]
@yuandanxu that seems unrelated, that error means that a test request was in the queue for a long time and therefore it was cancelled.

Could you please try again?

**Comment by yuandanxu** [comment]
@diemol yes, this issue still exists after remove the parameter. This issue doesn't exist before I pulled latest selenium. No tests can be started now. I will use previous version to have a test again.

**Comment by pearj** [comment]
I used the latest `elgalu/selenium` today as well and it worked fine too.

**Comment by WarleyGabriel** [comment]
Hi @diemol,

I used the latest version today.

`
elgalu/selenium     latest              6a177381ea4e        4 days ago          1.37GB
`
` 
dosel/zalenium      latest              894ba0c2f69f        11 days ago         793MB
`

Docker Command to start Zalenium:

`docker run -d -ti --name zalenium -p 4444:4444 -e ZALENIUM_EXTRA_JVM_PARAMS="-Dwebdriver.http.factory=apache" -v /var/run/docker.sock:/var/run/docker.sock  --privileged dosel/zalenium:latest start --timeZone "America/Sao_Paulo" --maxTestSessions 100 --desiredContainers 10`

It worked fine too.
I think the parameters have solved.

**Comment by Matrosskin** [comment]
@diemol Not sure if it is fine to create new issue so I ask here at first:
For me adding of `-e ZALENIUM_EXTRA_JVM_PARAMS="-Dwebdriver.http.factory=apache"` cause an error when i use the lates image dosel.zalenium (459bf04fbe39):
```
Docker version 17.12.1-ce, build 7390fc6
-- LOG 10:57:17:409086373 Ensuring docker works...
-- LOG 10:57:17:485645748 Ensuring no stale Zalenium related containers are still around...
-- LOG 10:57:17:501930025 Ensuring docker-selenium is available...
Copying files for Dashboard...
cp: cannot create regular file '/home/seluser/videos/index.html': No such file or directory
Starting Nginx reverse proxy...
Starting Selenium Hub...
Error: Could not find or load main class "-Dwebdriver.http.factory=apache"
...............................................................................................................................................................................................................................................................................................GridLauncher failed to start after 1 minute, failing...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   182  100   182    0     0  38930      0 --:--:-- --:--:-- --:--:-- 45500
<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx/1.10.3 (Ubuntu)</center>
</body>
</html>
```
Also tried with version 3.14.0b (2ea5f4d0ce1f) - without success. I did not try all versions, only found that version 3.12.0c (894ba0c2f69f) works well with your workaround.
@diemol could you please help with that error? Or would it be better to create separate issue?

**Comment by pearj** [comment]
@Matrosskin sounds like you have the syntax wrong somewhere. I suggest asking for help on slack and also pasting your full docker start command.

**Comment by kumar1210** [comment]
Hi Team, 

I am trying to run the zalenium using multiple machines(4), where in one of the machine i am running hub with the command : 
docker run --rm -ti --name zalenium --hostname zalenium_hub -p 4444:4444 -e ZALENIUM_EXTRA_JVM_PARAMS="-Dwebdriver.http.factory=apache" -v /var/run/docker.sock:/var/run/docker.sock -v /tmp/videos:/home/seluser/videos --privileged dosel/zalenium start --maxTestSessions 10 --maxDockerSeleniumContainers 5

and running the node in other machines with the below command, the number of nodes i am creating in each machine is equal to core in the machine : 
docker run -d --name `hostname`_node_0 -h `hostname`_node_0 -p 5550:5555 -e HUB_HOST=slc10gaq.us.oracle.com -e HUB_PORT=4444 -e REMOTE_HOST="http://`hostname --long`:5550" selenium/node-firefox ;


But still i am getting the error : 

[qtp109069556-89] ERROR o.o.g.w.s.handler.RequestHandler - cannot forward the request unexpected end of stream on Connection{slc12giq.us.oracle.com:5550, proxy=DIRECT hostAddress=slc12giq.us.oracle.com/10.250.106.145:5550 cipherSuite=none protocol=http/1.1}
java.io.IOException: unexpected end of stream on Connection{slc12giq.us.oracle.com:5550, proxy=DIRECT hostAddress=slc12giq.us.oracle.com/10.250.106.145:5550 cipherSuite=none protocol=http/1.1}
        at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:208)
        at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:88)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at org.openqa.selenium.remote.internal.OkHttpClient$Factory$1.lambda$createClient$1(OkHttpClient.java:152)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
        at okhttp3.RealCall.execute(RealCall.java:77)
        at org.openqa.selenium.remote.internal.OkHttpClient.execute(OkHttpClient.java:103)
        at org.openqa.grid.internal.TestSession.sendRequestToNode(TestSession.java:422)
        at org.openqa.grid.internal.TestSession.forward(TestSession.java:229)
        at org.openqa.grid.web.servlet.handler.RequestHandler.forwardRequest(RequestHandler.java:99)
        at org.openqa.grid.web.servlet.handler.RequestHandler.process(RequestHandler.java:133)
        at org.openqa.grid.web.servlet.DriverServlet.process(DriverServlet.java:85)
        at org.openqa.grid.web.servlet.DriverServlet.doPost(DriverServlet.java:69)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.seleniumhq.jetty9.servlet.ServletHolder.handle(ServletHolder.java:865)
        at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
        at io.prometheus.client.filter.MetricsFilter.doFilter(MetricsFilter.java:170)
        at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.seleniumhq.jetty9.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
        at org.seleniumhq.jetty9.security.SecurityHandler.handle(SecurityHandler.java:548)
        at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
        at org.seleniumhq.jetty9.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.seleniumhq.jetty9.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
        at org.seleniumhq.jetty9.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.seleniumhq.jetty9.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
        at org.seleniumhq.jetty9.server.handler.ContextHandler.doScope(ContextHandler.java:1242)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
        at org.seleniumhq.jetty9.server.handler.StatisticsHandler.handle(StatisticsHandler.java:174)
        at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.seleniumhq.jetty9.server.Server.handle(Server.java:503)
        at org.seleniumhq.jetty9.server.HttpChannel.handle(HttpChannel.java:364)
        at org.seleniumhq.jetty9.server.HttpConnection.onFillable(HttpConnection.java:260)
        at org.seleniumhq.jetty9.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
        at org.seleniumhq.jetty9.io.FillInterest.fillable(FillInterest.java:103)
        at org.seleniumhq.jetty9.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
        at org.seleniumhq.jetty9.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
        at org.seleniumhq.jetty9.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
        at org.seleniumhq.jetty9.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException: \n not found: limit=0 content=â€¦
        at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:237)
        at okhttp3.internal.http1.Http1Codec.readHeaderLine(Http1Codec.java:215)
        at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
        ... 61 common frames omitted
15:03:09.954 [Matcher thread] INFO  org.openqa.grid.internal.TestSlot - Trying to create a new session on test slot {server:CONFIG_UUID=7889c89e-10dd-4a35-8fe0-47e024c9d376, seleniumProtocol=WebDriver, browserName=firefox, maxInstances=1, moz:firefoxOptions={log={level=info}}, platformName=LINUX, version=69.0, applicationName=, platform=LINUX}
15:03:09.954 [Matcher thread] INFO  d.z.e.z.registry.ZaleniumRegistry - Test session with internal key 4168250e-d06e-4565-9b76-840d5c4cd5fd assigned to remote (http://slc12giq.us.oracle.com:5550) after 47 seconds (47812 ms).
15:03:25.493 [qtp109069556-77] ERROR o.o.g.w.s.handler.RequestHandler - cannot forward the request unexpected end of stream on Connection{slc15ahq.us.oracle.com:5551, proxy=DIRECT hostAddress=slc15ahq.us.oracle.com/10.247.152.59:5551 cipherSuite=none protocol=http/1.1}
java.io.IOException: unexpected end of stream on Connection{slc15ahq.us.oracle.com:5551, proxy=DIRECT hostAddress=slc15ahq.us.oracle.com/10.247.152.59:5551 cipherSuite=none protocol=http/1.1}
        at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:208)
        at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:88)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at org.openqa.selenium.remote.internal.OkHttpClient$Factory$1.lambda$createClient$1(OkHttpClient.java:152)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:126)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
        at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:200)
        at okhttp3.RealCall.execute(RealCall.java:77)
        at org.openqa.selenium.remote.internal.OkHttpClient.execute(OkHttpClient.java:103)
        at org.openqa.grid.internal.TestSession.sendRequestToNode(TestSession.java:422)
        at org.openqa.grid.internal.TestSession.forward(TestSession.java:229)
        at org.openqa.grid.web.servlet.handler.RequestHandler.forwardRequest(RequestHandler.java:99)
        at org.openqa.grid.web.servlet.handler.RequestHandler.process(RequestHandler.java:133)
        at org.openqa.grid.web.servlet.DriverServlet.process(DriverServlet.java:85)
        at org.openqa.grid.web.servlet.DriverServlet.doPost(DriverServlet.java:69)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.seleniumhq.jetty9.servlet.ServletHolder.handle(ServletHolder.java:865)
        at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1655)
        at io.prometheus.client.filter.MetricsFilter.doFilter(MetricsFilter.java:170)
        at org.seleniumhq.jetty9.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1642)
        at org.seleniumhq.jetty9.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
        at org.seleniumhq.jetty9.security.SecurityHandler.handle(SecurityHandler.java:548)
        at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
        at org.seleniumhq.jetty9.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.seleniumhq.jetty9.server.handler.ContextHandler.doHandle(ContextHandler.java:1340)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
        at org.seleniumhq.jetty9.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.seleniumhq.jetty9.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
        at org.seleniumhq.jetty9.server.handler.ContextHandler.doScope(ContextHandler.java:1242)
        at org.seleniumhq.jetty9.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
        at org.seleniumhq.jetty9.server.handler.StatisticsHandler.handle(StatisticsHandler.java:174)
        at org.seleniumhq.jetty9.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.seleniumhq.jetty9.server.Server.handle(Server.java:503)
        at org.seleniumhq.jetty9.server.HttpChannel.handle(HttpChannel.java:364)
        at org.seleniumhq.jetty9.server.HttpConnection.onFillable(HttpConnection.java:260)
        at org.seleniumhq.jetty9.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
        at org.seleniumhq.jetty9.io.FillInterest.fillable(FillInterest.java:103)
        at org.seleniumhq.jetty9.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
        at org.seleniumhq.jetty9.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
        at org.seleniumhq.jetty9.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
        at org.seleniumhq.jetty9.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
        at org.seleniumhq.jetty9.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException: \n not found: limit=0 content=â€¦
        at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:237)
        at okhttp3.internal.http1.Http1Codec.readHeaderLine(Http1Codec.java:215)
        at okhttp3.internal.http1.Http1Codec.readResponseHeaders(Http1Codec.java:189)
        ... 61 common frames omitted
15:03:25.597 [Matcher thread] INFO  org.openqa.grid.internal.TestSlot - Trying to create a new session on test slot {server:CONFIG_UUID=20cbf200-5a46-460e-bed9-1b44a75b0368, seleniumProtocol=WebDriver, browserName=firefox, maxInstances=1, moz:firefoxOptions={log={level=info}}, platformName=LINUX, version=69.0, applicationName=, platform=LINUX}


Any suggestions ?

**Comment by CrEaK** [comment]
Im getting the same error on my side. I thinks the error appears because since Selenium 3.141.0 the apache-client is removed. So the property "webdriver.http.factory=apache" is not working and selenium still using the okhttp-client and resulting in this errors :-(

**Comment by rootux** [comment]
I'm setting getting this error

**Comment by ssarraju** [comment]
This issue is happening even after adding those parameters.

**Comment by rakeshnambiar** [comment]
Hello all, any update or workaround for this?

I can see it's mentioned on the documentation `Known issues` section
https://opensource.zalando.com/zalenium/
